cirq/work/observable_measurement_data.py

```diff
@@ -0,0 +1,478 @@
+
+        all_obs_vals = np.array(
+            [
+                _obs_vals_from_measurements(
+                    bitstrings=self.bitstrings,
+                    qubit_to_index=self._qubit_to_index,
+                    observable=setting.observable,
+                    atol=atol,
+                )
+                for setting in self._simul_settings
+            ]
+        )
+
+        # https://github.com/numpy/numpy/issues/11502
+        if all_obs_vals.shape[0] == 1:
+            # coverage: ignore
+            cov = np.array([[np.var(all_obs_vals[0], ddof=1)]])
+            return cov
+
+        cov = np.cov(all_obs_vals, ddof=1) / all_obs_vals.shape[1]
+        return cov
+
+    def _validate_setting(self, setting: InitObsSetting, what: str):
+        mws = _max_weight_state([self.max_setting.init_state, setting.init_state])
+        mwo = _max_weight_observable([self.max_setting.observable, setting.observable])
+        if mws is None or mwo is None:
+            raise ValueError(
+                f"You requested the {what} for a setting that is not compatible "
+                f"with this BitstringAccumulator's meas_spec."
+            )
+
+    def variance(self, setting: InitObsSetting, *, atol: float = 1e-8):
+        """Compute the variance of the estimators of the given setting.
+
+        This is the normal variance divided by the number of samples to estimate
+        the certainty of our estimate of the mean. It is the standard error
+        of the mean, squared.
+
+        This uses `ddof=1` during the call to `np.var` for an unbiased estimator
+        of the variance in a hypothetical infinite population for consistency
+        with `BitstringAccumulator.covariance()` but differs from the default
+        for `np.var`.
+
+        Args:
+            setting: The setting
+            atol: The absolute tolerance for asserting coefficients are real.
+        """
+        if len(self.bitstrings) == 0:
+            # coverage: ignore
+            raise ValueError("No measurements")
+        self._validate_setting(setting, what='variance')
+
+        _, var = _stats_from_measurements(
+            bitstrings=self.bitstrings,
+            qubit_to_index=self._qubit_to_index,
+            observable=setting.observable,
+            atol=atol,
+        )
+
+        return var
+
+    def stderr(self, setting: InitObsSetting, *, atol: float = 1e-8):
+        """The standard error of the estimators for `setting`."""
+        return np.sqrt(self.variance(setting, atol=atol))
+
+    def means(self, *, atol: float = 1e-8) -> np.ndarray:
+        """Estimates of the means of the settings in this accumulator."""
+        return np.asarray([self.mean(setting, atol=atol) for setting in self.simul_settings])
+
+    def mean(self, setting: InitObsSetting, *, atol: float = 1e-8):
+        """Estimates of the mean of `setting`."""
+        if len(self.bitstrings) == 0:
+            # coverage: ignore
```

Comment:
Do you mind writing the test for these errors?

Reply:
done